src/meta/processors/job/JobManager.cpp

```diff
@@ -587,6 +587,34 @@ bool JobManager::checkJobExist(GraphSpaceID spaceId,
   return false;
 }
 
+nebula::cpp2::ErrorCode JobManager::checkNeedRecoverJobExist(GraphSpaceID spaceId,
+                                                             const cpp2::JobType& jobType) {
+  if (jobType == cpp2::JobType::DATA_BALANCE || jobType == cpp2::JobType::ZONE_BALANCE) {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto jobPre = MetaKeyUtils::jobPrefix(spaceId);
+    auto retCode = kvStore_->prefix(kDefaultSpaceId, kDefaultPartId, jobPre, &iter);
+    if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(INFO) << "Fetch jobs failed, error: " << apache::thrift::util::enumNameSafe(retCode);
+      return retCode;
+    }
+
+    for (; iter->valid(); iter->next()) {
+      if (!MetaKeyUtils::isJobKey(iter->key())) {
+        continue;
+      }
+      auto tup = MetaKeyUtils::parseJobVal(iter->val());
+      auto type = std::get<0>(tup);
+      auto status = std::get<2>(tup);
+      if (type == cpp2::JobType::DATA_BALANCE || type == cpp2::JobType::ZONE_BALANCE) {
+        if (status == cpp2::JobStatus::FAILED || status == cpp2::JobStatus::STOPPED) {
```

Comment:
how about remove the condition of STOPPED, because if the job always fail, we can't create a new one, so if a job failed and we don't want to continue, we stop it and create a new one, this way could be equal to "reset balance plan" in the past.

Reply:
make sense